src/storage/redis_db.cc

```diff
@@ -277,9 +277,8 @@ rocksdb::Status Database::Scan(const std::string &cursor, uint64_t limit, const
       keys->emplace_back(user_key);
       cnt++;
     }
-
     if (!storage_->IsSlotIdEncoded() || prefix.empty()) {
-      if (!keys->empty()) {
+      if (!keys->empty() && cnt >= limit) {
```

Comment:
I don't fully understand why a `cnt >= limit` is required?

Reply:
This code fix other bug. 

> Fixed a bug where the cursor did not return 0 when SCAN commands return less than number elements.

Reply:
Great! Would you mind add a testing for this?